api/src/pcapi/connectors/api_adresse.py

```diff
@@ -0,0 +1,189 @@
+            logger.error("No result from API Addresse for a municipality", extra={"queried_insee_code": insee_code})
+            raise NoResultException
+        return self._format_result(data)
+
+    def get_single_address_result(self, queried_address: str, insee_code: str) -> AddressInfo:
+        """
+        No human interaction so we limit to 1 result, and add a filter on the INSEE code (Code Officiel Géographique)
+        This will get the highest score result from the query, for a specific INSEE code.
+        An incorrect result would still be in the vicinity of the expected result, and can be later edited in pc pro
+        see https://forum.etalab.gouv.fr/t/interpretation-du-score/3852/4
+        If no result is found, we return the centroid of the municipality
+        """
+        parameters = {
+            "q": queried_address,
+            "citycode": insee_code,
+            "autocomplete": 0,
+            "limit": 1,
+        }
+
+        data = self._send_search_request(parameters=parameters)
+        if self._is_result_empty(data):
+            logger.warning(
+                "No result from API Addresse for queried address", extra={"queried_address": queried_address}
+            )
+            return self._get_municipality_centroid(insee_code=insee_code)
+
+        result = self._format_result(data)
+
+        extra = {
+            "id": result.id,
+            "label": result.label,
+            "latitude": result.latitude,
+            "longitude": result.longitude,
+            "queried_address": queried_address,
+            "score": result.score,
+        }
+        if result.score < RELIABLE_SCORE_THRESHOLD:
+            logger.warning("Result from API Addresse has a low score", extra=extra)
+        else:
+            logger.info("Retrieved details from API Adresse for query", extra=extra)
+        return result
+
+    def _get_coordinates_from_address_data(self, data: dict) -> _Geometry:
+        block = data["features"][0]["geometry"]["coordinates"]
+        return _Geometry(
+            latitude=block[0],
+            longitude=block[1],
+        )
+
+    def _get_debugging_info_from_address_data(self, data: dict) -> _Properties:
+        block = data["features"][0]["properties"]
+        return _Properties(
+            id=block["id"],
+            score=block["score"],
+            label=block["label"],
+        )
+
+    def _is_result_empty(self, result: dict) -> bool:
+        return len(result["features"]) == 0
+
+    def _format_result(self, data: dict) -> AddressInfo:
+        return AddressInfo(
+            id=self._get_debugging_info_from_address_data(data).id,
+            label=self._get_debugging_info_from_address_data(data).label,
+            latitude=self._get_coordinates_from_address_data(data).latitude,
+            longitude=self._get_coordinates_from_address_data(data).longitude,
+            score=self._get_debugging_info_from_address_data(data).score,
+        )
```

Comment:
(détail / question de goût) Je ne suis pas sûr de ce qu'apportent les fonctions `_get_debugging_info_from_address_data()`, `_get_coordinates_from_address_data()` et leur classes pydantic associées. La validation des données renvoyées par l'API ? Est-ce que ça vaut le coup ?

Le code me semble plus simple sans ces fonctions/classes, sans complexification dramatique de `_format_result()` : 

```suggestion
        properties = data["features"][0]["properties"]
        coordinates = data["features"][0]["geometry"]["coordinates"]
        return AddressInfo(
            id=properties["id"],
            label=properties["label"],
            latitude=coordinates["latitude"],
            longitude=coordinates["longitude"],
            score=properties["score"],
        )
```

Reply:
C'est vrai; je pensais à un moment ré-utiliser ces helpers, mais on peut s'en passer.